### PR TITLE
e2e/util.go: Move etcd spec related methods into e2eutil for common use

### PR DESCRIPTION
**Pull request by hasbro17**
Cluster spec methods grouped in `e2eutil` but mostly unchanged.

### CODE REVIEW

File: test/e2e/e2eutil/spec_util.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+import (
+	"strings"
+
+	"github.com/coreos/etcd-operator/pkg/spec"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewClusterSpec(genName string, size int) *spec.Cluster {
```
Comment (hongchaodeng) [inline test/e2e/e2eutil/spec_util.go]: NewCluster() or NewClusterTPR().

It was already a mistake.

**Comment by hasbro17** [comment]
@hongchaodeng PTAL

File: test/e2e/e2eutil/spec_util.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		Metadata: metav1.ObjectMeta{
+			GenerateName: genName,
+		},
+		Spec: spec.ClusterSpec{
+			Size: size,
+		},
+	}
+}
+
+func NewBackupPolicyS3() *spec.BackupPolicy {
```
Comment (hongchaodeng) [inline test/e2e/e2eutil/spec_util.go]: NewBackupPolicyS3 -> NewS3BackupPolicy

File: test/e2e/e2eutil/spec_util.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		BackupIntervalInSecond: 60 * 60,
+		MaxBackups:             5,
+		StorageType:            spec.BackupStorageTypeS3,
+		StorageSource: spec.StorageSource{
+			S3: &spec.S3Source{},
+		},
+	}
+}
+
+func NewBackupPolicyPV() *spec.BackupPolicy {
```
Comment (hongchaodeng) [inline test/e2e/e2eutil/spec_util.go]: NewBackupPolicyPV -> NewPVBackupPolicy

File: test/e2e/e2eutil/spec_util.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+		StorageType:            spec.BackupStorageTypePersistentVolume,
+		StorageSource: spec.StorageSource{
+			PV: &spec.PVSource{
+				VolumeSizeInMB: 512,
+			},
+		},
+	}
+}
+
+func EtcdClusterWithBackup(cl *spec.Cluster, backupPolicy *spec.BackupPolicy) *spec.Cluster {
```
Comment (hongchaodeng) [inline test/e2e/e2eutil/spec_util.go]: EtcdClusterWithBackup -> ClusterWithBackup

Same to others...

**Comment by hasbro17** [comment]
Also renamed `UpdateEtcdCluster`-->`UpdateCluster`
and `DeleteEtcdCluster`-->`DeleteCluster`

**Pull request merged**